packages/cli/src/commonlib/codeFlowLogin.ts

```diff
@@ -253,16 +251,31 @@ export class CodeFlowLogin {
   }
 
   async logout(): Promise<boolean> {
-    const accountCache = await loadAccountId(this.accountName);
-    if (accountCache) {
-      const dataCache = await this.msalTokenCache.getAccountByHomeId(accountCache);
-      if (dataCache) {
-        this.msalTokenCache?.removeAccount(dataCache);
+    const release = await this.mutex.acquire();
+    try {
+      const accountCache = await loadAccountId(this.accountName);
+      if (accountCache) {
+        const dataCache = await this.msalTokenCache.getAccountByHomeId(accountCache);
+        if (dataCache) {
+          this.msalTokenCache?.removeAccount(dataCache);
+        }
       }
+
+      await saveAccountId(this.accountName, undefined);
+      this.account = undefined;
+      return true;
+    } finally {
+      release();
     }
+  }
 
-    await saveAccountId(this.accountName, undefined);
-    return true;
+  // when using command account logout m365, activate will try to get region information at the start time.
+  // At the same time AccountLogout handler will try to logout account.
+  // This will trigger getStatus->logout account->getStatus process switching and cause silent getting token in logout status issue.
+  // This method provides a way to lock specific method and release to do other work.
+  public async getLock(): Promise<MutexInterface.Releaser> {
```

Comment:
Is it possible to use lock in `codeFlowLogin` only?
E.g., add lock to `reloadCache`, `getTokenByScope`, `logout`...

One best practice to use lock is to ensure the lock scope is as small as possible.

Reply:
Of course, another best practice is - do not use lock if possible :P

Reply:
I have tried to use lock in `codeFlowLogin` only, it does not address this problem. Because `getStatus` has used `reloadCache` and `getTokenByScope` methods. `logout` will insert the execution in the middle of two. I am trying to find a better way to fix this.